service/src/MSRIOGroup.cpp

```diff
@@ -1256,13 +1257,25 @@ namespace geopm
         }
         std::vector<std::unique_ptr<MSR> > msr_arr_custom;
         for (const auto &dir : plugin_paths) {
-            auto files = list_directory_files(dir);
-            for (const auto &file : files) {
-                std::string filename = dir + "/" + file;
-                if (string_begins_with(file, "msr_") && string_ends_with(file, ".json")) {
-                    data_files.insert(filename);
+            try {
```

Comment:
class SecurePath {
    SecurePath(const std::string &orig_path);
    SecurePath(const std::string &orig_path, int umask, bool is_same_user);
    ~SecurePath()
    std::string secure_path();
    int fid();
}

Reply:
SecurePath sp("foobar");
// replace all uses of "foobar" with sp.secure_path()